To: vim-dev@vim.org
Subject: Patch 7.3.021
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.021
Problem:    Conflict for defining Boolean in Mac header files.
Solution:   Define NO_X11_INCLUDES. (Rainer Muller)
Files:	    src/os_macosx.m, src/vim.h


*** ../vim-7.3.020/src/os_macosx.m	2010-08-15 21:57:29.000000000 +0200
--- src/os_macosx.m	2010-10-10 17:01:55.000000000 +0200
***************
*** 15,20 ****
--- 15,24 ----
      Error: MACOS 9 is no longer supported in Vim 7
  #endif
  
+ /* Avoid a conflict for the definition of Boolean between Mac header files and
+  * X11 header files. */
+ #define NO_X11_INCLUDES
+ 
  #include "vim.h"
  #import <Cocoa/Cocoa.h>
  
*** ../vim-7.3.020/src/vim.h	2010-08-15 21:57:29.000000000 +0200
--- src/vim.h	2010-10-10 17:02:56.000000000 +0200
***************
*** 192,199 ****
  #endif
  
  #ifdef NO_X11_INCLUDES
!     /* In os_mac_conv.c NO_X11_INCLUDES is defined to avoid X11 headers.
!      * Disable all X11 related things to avoid conflicts. */
  # ifdef FEAT_X11
  #  undef FEAT_X11
  # endif
--- 192,199 ----
  #endif
  
  #ifdef NO_X11_INCLUDES
!     /* In os_mac_conv.c and os_macosx.m NO_X11_INCLUDES is defined to avoid
!      * X11 headers.  Disable all X11 related things to avoid conflicts. */
  # ifdef FEAT_X11
  #  undef FEAT_X11
  # endif
*** ../vim-7.3.020/src/version.c	2010-10-09 17:21:42.000000000 +0200
--- src/version.c	2010-10-10 17:05:19.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     21,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
259. When you enter your name in the AltaVista search engine, the top ten
     matches do indeed refer to you.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
